contextid_test.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func getIdFromRequest(t *testing.T, server *httptest.Server) uint {
+	res, err := http.Get(server.URL)
+	assert.NoError(t, err)
```

Comment:
I like the idea of using `require` in test cases where an error indicates the test itself is broken rather than the unit under test. I will re-work my tests to use `require` in those circumstances.

Reply:
Fixed in 719352970088b7beb366fa5b4c354ba42e672213, 9ddd9a5c4026641553706ace54060736c6f13150 and b174e824ff71a4e78e82f937477c7ad426b59f91. Will rebase-squash prior to merging.

Reply:
bit late to the party, sorry. An alternative that is a bit more explicitly (if you don't know the difference between testify's `require` and `assert` off the top of your head) and to be used if needed rarely is: if testbroken { [assert.FailNow()](https://godoc.org/github.com/stretchr/testify/assert#FailNow)  }`  

Aside: you can force require behaviour for _all_ assert statements with `go test -failfast` I believe.